devo/icon_devo/util/api.py

```diff
@@ -0,0 +1,146 @@
+                cause=f"{time_string} does not appear to be a valid time",
+                assistance="Please check your step inputs and try again.",
+                data=e,
+            )
+
+        if not parsed_time:
+            raise PluginException(
+                cause=f"{time_string} does not appear to be a valid time",
+                assistance="Please check your step inputs and try again.",
+            )
+
+        return int(parsed_time.timestamp())
+
+    def _post_to_api(self, endpoint: str, post_body: dict) -> dict:
+        self.logger.info(f"Starting post to API: {endpoint}")
+
+        # Need this because if it starts with / url join thinks it's a root
+        url = urllib.parse.urljoin(self.base_url, endpoint.lstrip("/"))
+        try:
+            response = self.session.post(url=url, json=post_body, headers=self.headers, stream=True)
+
+            if response.status_code == 400:
+                raise PluginException(
+                    cause=f"API Error. API returned {response.status_code}",
+                    assistance="Bad request or invalid JSON.",
+                    data=response.text,
+                )
+            if response.status_code > 401 and response.status_code < 404:
+                raise PluginException(PluginException.Preset.UNAUTHORIZED, data=response.text)
+            if response.status_code == 404:
+                raise PluginException(
+                    cause=f"API Error. API returned {response.status_code}",
+                    assistance=f"The object at {url} does not exist.",
+                    data=response.text,
+                )
+            # Success; no content
+            if response.status_code == 204:
+                return None
+
+            if response.status_code >= 200 and response.status_code < 300:
+                self.logger.info("Returning from post to API")
+                output = self._get_stream(response)
+                return output
+
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=response.text)
+        except json.decoder.JSONDecodeError as e:
+            self.logger.info(f"Invalid json: {e}")
+            raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=f"Response was:\n{response.text}")
+        except requests.exceptions.HTTPError as e:
+            self.logger.info(f"Request to {url} failed: {e}")
+            raise PluginException(preset=PluginException.Preset.UNKNOWN)
+
+    def _get_stream(self, request: requests.Request) -> dict:
+        content = ""
+        for chunk in request.iter_content(2048):
+            content += chunk.decode()
+            if len(content) > RESPONSE_MAXSIZE:
+                request.close()
+                raise PluginException(
+                    cause="Response too large.",
+                    assistance="The response recieved from the server was too large, please edit your input to reduce the returned data size and try again.",
```

Comment:
```suggestion
                    assistance="The response received from the server was too large, please edit your input to reduce the returned data size and try again.",
```

It would be a good idea to tell the user to reference the Troubleshooting section of the docs that points out ways to reduce the data size returned.